src/controllers/userController.js

```diff
@@ -0,0 +1,58 @@
+import { createUserOnDb } from "../services/userService.js";
+import { ipAddressModel } from "../models/ipAddressModel.js";
+
+export const registerUser = async (req, res) => {
+    const ipAddress = req.headers['x-forwarded-for'] || req.socket.remoteAddress;
+
+    const { name, email } = req.body || {};
+
+        try {
+            const ipExists = await ipAddressModel.findOne({ ipAddress: ipAddress });
+
+            if(ipExists === null) {
+                return res.status(403).send({
+                    message: "IP not registered"
+                });
+            }
+
+            if(!name || !email) {
+                const updatedResult = await ipAddressModel.updateOne(
+                    {
+                        ipAddress: ipAddress,
+                        token: { $gt: 0 }
+                    },
+                    {
+                        $inc: { token: -1 }
+                    }    
+                );
+
+                if(updatedResult.matchedCount === 0) {
+                    return res.status(403).send({
+                    message: "You do not have enough tokens."
+                });
+            }
+                return res.status(401).send({
+                    message: "You must provide the fields to create an user"
+                });
+            }
+
+
+            const newUser = await createUserOnDb(name, email, session)
```

Comment:
o token só atualiza quando a requisição não manda `name` e `email`?

eu imagino que o token seja por request, certo? aqui está sendo somente quando o request não possui as informações necessárias (se estiver correto dessa forma, desconsidere o review)

de qualquer forma, seria legal ter uma função separada para pegar o ip com token maior que 0 e atualizar o token, assim como vc tem services pra criar usuário no db. na minha cabeça ficaria algo assim: 
```suggestion
            if(!name || !email) {
                return res.status(401).send({
                    message: "You must provide the fields to create an user"
                });
            }
            const updatedResult = await validateTokenAndDecreaseIfExists();

            if(updatedResult.matchedCount === 0) {
                    return res.status(403).send({
                    message: "You do not have enough tokens."
                });
            }

            const newUser = await createUserOnDb(name, email, session)
```